keanu-project/src/main/java/io/improbable/keanu/distributions/discrete/Binomial.java

```diff
@@ -50,29 +50,22 @@ private static int sample(double p, int n, KeanuRandom random) {
     public DoubleTensor logProb(IntegerTensor k) {
         DoubleTensor logBinomialCoefficient = getLogBinomialCoefficient(k, n);
 
-        DoubleTensor logBinomial = p.pow(k.toDouble())
-            .times(
-                p.unaryMinus().plusInPlace(1.0).powInPlace(n.minus(k).toDouble())
-            ).logInPlace();
+        DoubleTensor kDouble = k.toDouble();
+        DoubleTensor nDouble = n.toDouble();
+        DoubleTensor logPPowK = kDouble.times(p.log());
+        DoubleTensor log1MinusPPowNMinusK = nDouble.minus(kDouble).timesInPlace(p.unaryMinus().plusInPlace(1.0).logInPlace());
```

Comment:
I'd find it clearer split over 3 lines:

        DoubleTensor nMinusK = nDouble.minus(kDouble);
        DoubleTensor logOneMinusP = p.unaryMinus().plusInPlace(1.0).logInPlace();
        DoubleTensor nMinusKlogOneMinusP = nMinusK.timesInPlace(logOneMinusP);

Reply:
Agreed.